pandas/tests/tools/test_to_datetime.py

```diff
@@ -2530,3 +2530,16 @@ def test_empty_string_datetime_coerce__unit():
     # verify that no exception is raised even when errors='raise' is set
     result = to_datetime([1, ""], unit="s", errors="raise")
     tm.assert_index_equal(expected, result)
+
+
+def test_to_datetime_monotonic_increasing_index():
+    # GH28238
+    # Create date range of 1000 hour periods
+    times = date_range(datetime.now(), periods=1000, freq="h")
+    # Random sort
+    times = times.to_frame(index=False, name="DT").sample(1000)
+    # Divide index integers by 1000
+    times.index = times.index.to_series().astype(float) / 1000
+    # Convert to datetime
+    result = to_datetime(times.iloc[:, 0])
+    assert result.values.dtype == np.dtype("datetime64[ns]")
```

Comment:
Could you try creating a test where you can compare the actual values?

Reply:
Just updated it, along with np.ravel to ensure that the values before the conversion could be compared via 1-d arrays.